### ISSUE TITLE
Add a tool function to deal with $ASPECT_SOURCE_DIR

### ISSUE THREAD

**Issue by bangerth** [geodynamics/aspect#666]
We have a whole bunch of places that now do this:

```
          const std::string      subst_text = "$ASPECT_SOURCE_DIR";
          std::string::size_type position;
          while (position = data_directory.find (subst_text),  position!=std::string::npos)
            data_directory.replace (data_directory.begin()+position,
                                    data_directory.begin()+position+subst_text.size(),
                                    ASPECT_SOURCE_DIR);
```

This would make for a good function in `utilities.cc`.

**Comment by bangerth** [comment]
This is the list of places that right now would call this function:
```
> grep -r 'ASPECT_SOURCE_DIR"' source/
source/main.cc:                                                     "$ASPECT_SOURCE_DIR",
source/utilities.cc:                                                      "$ASPECT_SOURCE_DIR",
source/initial_conditions/SAVANI_perturbation.cc:                                                       "$ASPECT_SOURCE_DIR",
source/initial_conditions/solidus.cc:                                                          "$ASPECT_SOURCE_DIR",
source/initial_conditions/adiabatic_boundary.cc:                                                        "$ASPECT_SOURCE_DIR",
source/initial_conditions/S40RTS_perturbation.cc:                                                       "$ASPECT_SOURCE_DIR",
source/material_model/depth_dependent.cc:                                                                       "$ASPECT_SOURCE_DIR",
source/material_model/steinberger.cc:                                                       "$ASPECT_SOURCE_DIR",
source/velocity_boundary_conditions/gplates.cc:                                                        "$ASPECT_SOURCE_DIR",
source/particle/generator/ascii_file.cc:                                                              "$ASPECT_SOURCE_DIR",
```

**Comment by anne-glerum** [comment]
All occurrences already seem replaced by a dealii utilities function #853 :
      `  data_directory = Utilities::replace_in_string(prm.get ("Data directory"),
                                                      "$ASPECT_SOURCE_DIR",
                                                      ASPECT_SOURCE_DIR);`

**Comment by bangerth** [comment]
@anne-glerum -- yes. I was more thinking of just replacing all of `Utilities::replace_in_string(prm.get ("Data directory"), "$ASPECT_SOURCE_DIR", ASPECT_SOURCE_DIR);` by one call like `Utilities::expand_ASPECT_SOURCE_DIR(prm.get("Data directory"));`

**Comment by class4kayaker** [comment]
On it

**Comment by gassmoeller** [comment]
Fixed by #1072.
